### ISSUE TITLE
[enhancement] Mouse Controls?

### ISSUE THREAD

**Issue by Norbtails97** [emulatorjs/emulatorjs#816]
I can put keyboard controls on the emulator, but what about using the mouse? If I want to play Duck Hunt, I need the zapper controls. If I want to play Mario Paint, I need the SNES Mouse. Mabe if you ADD a easy, user-friendly mouse controls to emulate the mouse/Light Gun controls; that will be nice.

**Comment by michael-j-green** [comment]
@ethanaobrien; we might need to revisit that discussion on mouse capture.

Thinking we do this:
* provide an option to enable mouse capture on canvas click - this option should have a corresponding variable exposed to the host page (default false)
* if that first option is enabled, then provide another option to enable auto-capture - this option should have a corresponding variable exposed to the host page (default false)

Mouse capture is a requirement for many platforms, and mouse capture means that we don't have to do any funky scaling to match the host mouse movements to the emulation.

**Comment by ethanaobrien** [comment]
Yeah that works, we do also need to pass device ids to retroarch for lightgun/mouse related items too in the end

**Comment by michael-j-green** [comment]
Great :) I'll build out the mouse capture functions this week, but I'll have to have a chat with you around the device id component.

**Comment by michael-j-green** [comment]
Work to complete will be in #788

**Comment by ethanaobrien** [comment]
Hey @Norbtails97   is this still an issue? Duck hunt should be fixed, not sure about mario paint

**Comment by Lunascaped** [comment]
> Hey [@Norbtails97](https://github.com/Norbtails97) is this still an issue? Duck hunt should be fixed, not sure about mario paint

Mario paint doesn't seem to work
